File: src/common/web/settings/GeneralSettingIDs.ts
import { SettingID } from "../utils/config/SettingID";

/**
 * Identifiers for general settings.
 *
 * @property Debug - Whether debug mode should be enabled (value type: ``bool``).
 * @property VerboseNotifications - Whether to display more verbose notifications. (value type: ``bool``).
 * @property NotificationTimeout - The timeout for overlay notifications in seconds (value type: ``number``).
 */
export class GeneralSettingIDs {
    public static readonly Debug = new SettingID("general", "debug");

    public static readonly VerboseNotifications = new SettingID("general", "verbose_notifications");
    public static readonly NotificationTimeout = new SettingID("general", "notification_timeout");
}

File: src/common/py/settings/general_setting_ids.py
from ..utils.config import SettingID


class GeneralSettingIDs:
    # pylint: disable=too-few-public-methods
    """
    Identifiers for general settings.

    Attributes:
        DEBUG: Whether debug mode should be enabled (value type: ``bool``).
        DEBUG_TRACE: Whether to trace function calls (value type: ``bool``).
    """
    DEBUG = SettingID("general", "debug")
    DEBUG_TRACE = SettingID("general", "debug_trace")
